ctru-rs/examples/ir-user.rs

```diff
@@ -0,0 +1,251 @@
+        self.bottom_console.select();
+    }
+
+    fn connect_to_cpp(&self, hid: &Hid) -> ConnectionResult {
+        // Connection loop
+        loop {
+            hid.scan_input();
+            if hid.keys_held().contains(KeyPad::KEY_START) {
+                return ConnectionResult::Canceled;
+            }
+
+            // Start the connection process
+            self.ir_user
+                .require_connection(IrDeviceId::CirclePadPro)
+                .expect("Couldn't initialize circle pad pro connection");
+
+            // Wait for the connection to establish
+            if let Err(e) = self
+                .connection_status_event
+                .wait_for_event(Duration::from_millis(100))
+            {
+                if !e.is_timeout() {
+                    panic!("Couldn't initialize circle pad pro connection: {e}");
+                }
+            }
+
+            self.print_status_info();
+            if self.ir_user.get_status_info().connection_status == 2 {
+                println!("Connected!");
+                break;
+            }
+
+            // If not connected (ex. timeout), disconnect so we can retry
+            self.ir_user
+                .disconnect()
+                .expect("Failed to disconnect circle pad pro connection");
+
+            // Wait for the disconnect to go through
+            if let Err(e) = self
+                .connection_status_event
+                .wait_for_event(Duration::from_millis(100))
+            {
+                if !e.is_timeout() {
+                    panic!("Couldn't initialize circle pad pro connection: {e}");
+                }
+            }
+        }
+
+        // Sending first packet retry loop
+        loop {
+            hid.scan_input();
+            if hid.keys_held().contains(KeyPad::KEY_START) {
+                return ConnectionResult::Canceled;
+            }
+
+            // Send a request for input to the CPP
+            if let Err(e) = self
+                .ir_user
+                .request_input_polling(CPP_CONNECTION_POLLING_PERIOD_MS)
+            {
+                println!("Error: {e:?}");
+            }
+            self.print_status_info();
+
+            // Wait for the response
+            let recv_event_result = self
+                .receive_packet_event
+                .wait_for_event(Duration::from_millis(100));
+            self.print_status_info();
+
+            if recv_event_result.is_ok() {
+                println!("Got first packet from CPP");
+                self.handle_packets();
+                break;
+            }
+
+            // We didn't get a response in time, so loop and retry
+        }
+
+        ConnectionResult::Connected
+    }
+
+    fn handle_packets(&self) {
+        let packets = self.ir_user.get_packets();
+        let packet_count = packets.len();
+        let Some(last_packet) = packets.last() else { return };
+
+        // Use a buffer to avoid flickering the screen (write all output at once)
```

Comment:
you might also be able to simply turn on double buffering for whichever screen this is getting printed to